substrate/frame/staking-async/src/tests/slashing.rs

```diff
@@ -683,18 +677,8 @@ fn slashing_nominators_by_span_max() {
 		let slash_1_amount = Perbill::from_percent(10) * nominated_value_11;
 		assert_eq!(asset::stakeable_balance::<T>(&101), 500 - slash_1_amount);
 
-		let expected_spans = vec![
-			slashing::SlashingSpan { index: 1, start: 4, length: None },
-			slashing::SlashingSpan { index: 0, start: 0, length: Some(4) },
-		];
-
-		let get_span = |account| SlashingSpans::<T>::get(&account).unwrap();
-
-		assert_eq!(get_span(11).iter().collect::<Vec<_>>(), expected_spans);
-		assert_eq!(get_span(101).iter().collect::<Vec<_>>(), expected_spans);
-
-		// second slash: higher era, higher value, same span.
-		add_slash_in_era_with_value(21, 3, Perbill::from_percent(30));
```

Comment:
Similarly, if I were you I would add some events here as well. 

In general I have learned to like events to be in fact superior than asserting on a lot of storage items. I generally always assert on events, and only if really needed add storage assertions. 

I can see though that both are necessary.

Reply:
added